python-backend/app/api/parties/party/models/party.py

```diff
@@ -69,15 +65,7 @@ def json(self, show_mgr=True, relationships=[]):
             self.party_name,
             'name':
             self.name,
-            'address': [{
-                'suite_no': self.suite_no,
-                'address_line_1': self.address_line_1,
-                'address_line_2': self.address_line_2,
-                'city': self.city,
-                'sub_division_code': self.sub_division_code,
-                'post_code': self.post_code,
-                'address_type_code': self.address_type_code
-            }]
+            'address': self.address[0].json() if len(self.address) > 0 else [{}]
```

Comment:
empty lists are falsy, don't need to do a length check. should this return empty list? or list with empty object.

Reply:
I'm strongly against coercing an empty list. In python, an empty list is false. In Javascript, an empty array is truthy. Considering those are the two big languages in this repo, I think we should avoid that conflicting edge case

As far as the return value, the entire `json` method is all for backwards compatibility. Since every party had exactly one address, and it was in the form of an object as the first item in an array, I went with this format `[{}]`. I updated the frontend to handle an empty array where I noticed it, but this was the safer choice, imo.

Reply:
We have run across this a few times, but the backend should be treated as a separate application that is trying to be the best version of itself, not to accommodate or make it easier for the current FE.

Reply:
Yes. I agree. Perhaps I wasn't clear enough... the `json` method is strictly for backwards compatibility within the backend itself (mine_party_appt uses it). This will be removed as soon as that dependency is gone. The reason for keeping an empty object as the first item in the list is to avoid a potential runtime exception if I missed some reference to the first item in the address response because it was previously always present as the first item in the list.

I don't see how this has any relation to accommodating the frontend. It's backwards compatibility with the existing API & with the other parts of the backend codebase.

Reply:
Ah you have handled the frontend to handle empty array, ok great. If the JSON method is to be removed later, than all is good. Thanks for clarifying.